packages/flutter/lib/src/widgets/banner.dart

```diff
@@ -178,64 +178,32 @@ class BannerPainter extends CustomPainter {
   bool hitTest(Offset position) => false;
 
   double _translationX(double width) {
-    switch (layoutDirection) {
-      case TextDirection.rtl:
-        switch (location) {
-          case BannerLocation.bottomEnd:
-            return _kBottomOffset;
-          case BannerLocation.topEnd:
-            return 0.0;
-          case BannerLocation.bottomStart:
-            return width - _kBottomOffset;
-          case BannerLocation.topStart:
-            return width;
-        }
-      case TextDirection.ltr:
-        switch (location) {
-          case BannerLocation.bottomEnd:
-            return width - _kBottomOffset;
-          case BannerLocation.topEnd:
-            return width;
-          case BannerLocation.bottomStart:
-            return _kBottomOffset;
-          case BannerLocation.topStart:
-            return 0.0;
-        }
-    }
+    return switch ((layoutDirection, location)) {
+      (TextDirection.rtl, BannerLocation.topEnd)      => 0.0,
+      (TextDirection.ltr, BannerLocation.topStart)    => 0.0,
```

Comment:
Should this be:
```suggestion
      (_, BannerLocation.topEnd)      => 0.0,
```

Reply:
I wrote it out that way since the [Flutter Style Guide](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-using-if-chains-or--or--with-enum-values) says to avoid default cases with `enum` values.

The `TextDirection` enum probably won't ever be expanded upon, but IMO it'd still be nice to follow the convention.

Reply:
Yeah, grouping by right-hand value wasn't the best idea. Hopefully it's much less confusing now!